eng/scripts/update-pathversions.rs

```diff
@@ -30,11 +40,19 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
     let package_versions = get_package_versions(&toml_files);
 
     for (path, mut document) in toml_files {
-        update_package_versions(document.as_table_mut(), &package_versions);
+        let is_publishable_package = document
+            .get("package")
+            .and_then(Item::as_table)
+            .and_then(|table| table.get("publish").and_then(Item::as_bool))
+            .unwrap_or(true);
+
+        let should_add = is_publishable_package && mode == "add";
```

Comment:
Something is wrong if this added the `version`s above.

Reply:
fixed